cycles-ledger/tests/tests.rs

```diff
@@ -930,59 +947,917 @@ fn test_withdraw_fails() {
+    let account1 = account(1, None);
+    let account1_1 = account(1, Some(1));
+    let account1_2 = account(1, Some(2));
+    let account1_3 = account(1, Some(3));
+    let account1_4 = account(1, Some(4));
+    let withdrawer1 = account(102, None);
+    let withdrawer1_1 = account(102, Some(1));
+    let withdraw_receiver = env.state_machine.create_canister(None);
+
+    // make deposits to the user and check the result
+    let deposit_res = env.deposit(account1, 1_000_000_000, None);
+    assert_eq!(deposit_res.block_index, 0_u128);
+    assert_eq!(deposit_res.balance, 1_000_000_000_u128);
+    let _deposit_res = env.deposit(account1_1, 1_000_000_000, None);
+    let _deposit_res = env.deposit(account1_2, 1_000_000_000, None);
+    let _deposit_res = env.deposit(account1_3, 1_000_000_000, None);
+    let _deposit_res = env.deposit(account1_4, 1_000_000_000, None);
+    let mut expected_total_supply = 5_000_000_000;
+    assert_eq!(env.icrc1_total_supply(), expected_total_supply);
+
+    // withdraw cycles from main account
+    let withdraw_receiver_balance = env.state_machine.cycle_balance(withdraw_receiver);
+    let withdraw_amount = 500_000_000_u128;
+    env.icrc2_approve_or_trap(
+        account1.owner,
+        ApproveArgs {
+            from_subaccount: None,
+            spender: withdrawer1,
+            amount: Nat::from(withdraw_amount + FEE),
+            expected_allowance: None,
+            expires_at: None,
+            fee: None,
+            memo: None,
+            created_at_time: None,
+        },
+    );
+    let withdraw_idx = env.withdraw_from_or_trap(
+        withdrawer1.owner,
+        WithdrawFromArgs {
+            from: account1,
+            to: withdraw_receiver,
+            created_at_time: None,
+            amount: Nat::from(withdraw_amount),
+            spender_subaccount: withdrawer1.subaccount,
+        },
+    );
+    assert_eq!(
+        withdraw_receiver_balance + withdraw_amount,
+        env.state_machine.cycle_balance(withdraw_receiver)
+    );
+    assert_eq!(
+        env.icrc2_allowance(account1, withdrawer1),
+        Allowance {
+            allowance: 0_u128.into(),
+            expires_at: None
+        }
+    );
+    assert_eq!(
+        env.icrc1_balance_of(account1),
+        1_000_000_000 - withdraw_amount - FEE - FEE
+    );
+    expected_total_supply -= withdraw_amount + FEE + FEE;
+    assert_eq!(env.icrc1_total_supply(), expected_total_supply);
+
+    // check that the burn block created is correct
+    assert_display_eq(
+        &env.get_block(withdraw_idx.clone()),
+        &Block {
+            // The new block parent hash is the hash of the last deposit.
+            phash: Some(env.get_block(withdraw_idx - 1u8).hash().unwrap()),
+            // The effective fee of a burn block created by a withdrawal
+            // is the fee of the Ledger. This is different from burn in
+            // other Ledgers because the operation transfers cycles.
```

Comment:
```suggestion
            // is the fee of the ledger. This is different from burn in
            // other ledgers because the operation transfers cycles.
```